external/fv3fit/fv3fit/reservoir/readout.py

```diff
@@ -88,31 +104,32 @@ def dump(self, path: str) -> None:
         """
         # convert to sparse before saving so a single dump/load can be used
         with fsspec.open(os.path.join(path, self._COEFFICIENTS_NAME), "wb") as f:
-            scipy.sparse.save_npz(f, scipy.sparse.coo_matrix(self.coefficients))
+            np.save(f, self.coefficients, allow_pickle=False)
         with fsspec.open(os.path.join(path, self._INTERCEPTS_NAME), "wb") as f:
-            np.save(f, self.intercepts)
+            np.save(f, self.intercepts, allow_pickle=False)
 
     @classmethod
     def load(cls, path: str) -> "ReservoirComputingReadout":
         with fsspec.open(os.path.join(path, cls._COEFFICIENTS_NAME), "rb") as f:
-            coefficients = scipy.sparse.load_npz(f)
+            coefficients = np.load(f)
         with fsspec.open(os.path.join(path, cls._INTERCEPTS_NAME), "rb") as f:
             intercepts = np.load(f)
         return cls(coefficients=coefficients, intercepts=intercepts)
 
 
-def combine_readouts(readouts: Sequence[ReservoirComputingReadout]):
-    coefs, intercepts = [], []
-    for readout in readouts:
-        coefs.append(readout.coefficients)
-        intercepts.append(readout.intercepts)
-
-    # Merge the coefficient arrays of individual readouts into single
-    # block diagonal matrix
-    combined_coefficients = scipy.sparse.block_diag(coefs)
+def combine_readouts_from_subdomain_regressors(
```

Comment:
Switched to take in regressors so that all of the translation to a combined readout can be located here.